homeassistant/components/ecowitt/sensor.py

```diff
@@ -273,8 +343,31 @@ def __init__(
         """Initialize the sensor."""
         super().__init__(sensor)
         self.entity_description = description
+        self.restored_value: StateType | datetime | date | Decimal = None
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity about to be added to hass."""
+        await super().async_added_to_hass()
+        if self._attr_native_value is not None:
+            return
+
+        if (sensor_data := await self.async_get_last_sensor_data()) is not None:
+            self._attr_native_value = sensor_data.native_value
+            self._attr_available = True
+            self.restored_value = sensor_data.native_value
 
     @property
-    def native_value(self) -> StateType | datetime:
+    def native_value(self) -> StateType | datetime | date | Decimal:
         """Return the state of the sensor."""
-        return self.ecowitt.value
+        if self.ecowitt.value is not None:
+            return self.ecowitt.value
+        return self.restored_value
+
+    @property
+    def available(self) -> bool:
+        """Return whether the state is based on actual reading from device."""
+        if self.ecowitt.value is not None:
+            return (self.ecowitt.last_update_m + 5 * 60) > time.monotonic()
+        if self.restored_value:
+            return True
```

Comment:
What happens if you stop HA, unplug the Ecowitt base station, and restart HA?
Will all entities be falsely always available?

Reply:
You're right. 
In https://github.com/home-assistant/core/pull/134442/commits/8d964a20ce962d1f09ee0842d54a9f1cd25e766e
I implemented a change. This would probably after the second restart make the entity unavailable.